### PR TITLE
feat(python): Add storage_options in `write_parquet`

### PR DESCRIPTION
**Pull request by ion-elgreco**
https://github.com/pola-rs/polars/issues/9770

You can now pass fsspec kwargs into write_parquet just as in read_parquet.

Example:

```python
import polars as pl

df = pl.from_repr(
"""
┌─────┬─────┬─────┬────────┐
│ a   ┆ b   ┆ c   ┆ column │
│ --- ┆ --- ┆ --- ┆ ---    │
│ i64 ┆ i64 ┆ i64 ┆ str    │
╞═════╪═════╪═════╪════════╡
│ 1   ┆ 2   ┆ 3   ┆ hello  │
│ 2   ┆ 1   ┆ 3   ┆ world  │
│ 1   ┆ 2   ┆ 2   ┆ ...    │
└─────┴─────┴─────┴────────┘
""")

fp = "abfs://polarstest@<storage_account_name>.dfs.core.windows.net/test/new.parquet"
storage_options = {
    "account_key": <KEY>,
}
df.write_parquet(fp, storage_options=storage_options)

pl.read_parquet(fp, storage_options=storage_options)

```

<html>
<body>
<small style="color: rgb(212, 212, 212); font-family: &quot;Segoe WPC&quot;, &quot;Segoe UI&quot;, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">shape: (3, 4)</small>

a | b | c | column
-- | -- | -- | --
i64 | i64 | i64 | str
1 | 2 | 3 | "hello"
2 | 1 | 3 | "world"
1 | 2 | 2 | "..."

</body>
</html>

### CODE REVIEW

File: py-polars/polars/io/_utils.py [hunk 1]
```diff
@@ -193,3 +193,77 @@ def _process_http_file(path: str, encoding: str | None = None) -> BytesIO:
...
+                if infer_storage_options(file)["protocol"] == "file":
+                    return managed_file(normalise_filepath(file, False))
+            kwargs["mode"] = "wb"
+            kwargs["encoding"] = encoding
+            return fsspec.open(file, **kwargs)
+
+    if isinstance(file, str):
+        file = normalise_filepath(file, False)
+        if has_non_utf8_non_utf8_lossy_encoding:
+            with open(file, encoding=encoding_str) as f:
```
Comment (cnpryer) [inline py-polars/polars/io/_utils.py:264]:
```
polars/io/_utils.py:264:18: PTH123 `open()` should be replaced by `Path.open()`
```

Comment (ion-elgreco) [reply py-polars/polars/io/_utils.py]: @cnpryer in the mergeconflict, I saw that Path(file).open() was added, so I changed to that.

**Comment by ion-elgreco** [comment]
@stinodego @cnpryer are there any changes I need to make for this PR to get merged?

**Review by svaningelgem — COMMENTED** [review]
Hi @ion-elgreco : I'd love to see the possibility to pass in the path directly, much as it is being done in the read_parquet. After that it's just opened with fsspec.

Additionally: why only parquet, why not the other write_* methods as well?

**Comment by ion-elgreco** [comment]
> Hi @ion-elgreco : I'd love to see the possibility to pass in the path directly, much as it is being done in the read_parquet. After that it's just opened with fsspec.
> 
> Additionally: why only parquet, why not the other write_* methods as well?

Was checking write_parquet at that time and that seemed easiest to add. I could add the other ones, but I need to know if I'm on the right track first^^

**Pull request closed without merging**